includes/wikia/api/DesignSystemApiController.class.php

```diff
@@ -195,4 +203,60 @@ private function isWikiaOrgCommunity() {
 
 		return wfGetBaseDomainForHost( parse_url( $wgServer, PHP_URL_HOST ) ) === $wgWikiaOrgBaseDomain;
 	}
+
+
+	/**
+	 * External API request to IntentX
+	 */
+	public function getFandomShopDataFromIntentX() {
+		global $wgCityId, $wgMemc, $wgFandomShopMap, $wgFandomShopUrl;
+
+		// get id from parameter or default to city id
+		$id = $this->getVal( static::PARAM_ID, $wgCityId );
+
+		$memcKey = wfMemcKey( self::MEMC_PREFIX_FANDOM_STORE, $id );
+		$slicePos = strpos( $memcKey, ':' );
+		// remove previx from memckey
+		$memcKey = substr( $memcKey, $slicePos + 1 );
+
+		// don't make api call if not fandom store community
+		if ( !array_key_exists( $id, $wgFandomShopMap ) ) {
+			return null;
+		}
+
+		// do api request
+		$client = new Client( [
+			'base_uri' => $wgFandomShopUrl,
+			'timeout' => 30.0
+		] );
+		$params = [
+			'clientId' => 'fandom',
+			'relevanceKey' => $wgFandomShopMap[ $id ],
+		];
+
+		try {
+			$response = $client->get( '', [
+				'query' => build_query( $params, PHP_QUERY_RFC1738 ),
+			] );
+			$body = $response->getBody();
+			$data = json_decode( $body );
+
+			// if valid json store in cache and return
+			if ( $data && $data->results ) {
+				$wgMemc->set( $memcKey, $data, self::TTL_INFINITE );
+				return $this->setResponseData( $data->results );
+			} 
+			// if no data return empty array
+			return $this->setResponseData( [] );
```

Comment:
We should log this state too.

Reply:
Added logging here. "unable to retrieve valid response data".